devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,552 @@
+  parameter.parameter5title = "Motion inactive report time [Minutes] after active"
+  parameter.parameter5description = ""
+  parameter.parameter5type = "number"
+  parameter.parameter5options = "1..127"
+  parameter.parameter5enumMap = []
+  parameter.parameter5default = 3
+  parameter.parameter5size = 1
+  
+  parameter.parameter6name = "MotionSensitivity"
+  parameter.parameter6title = "Motion active sensitivity"
+  parameter.parameter6description = ""
+  parameter.parameter6type = "enum"
+  parameter.parameter6options = ["Highest", "Higher", "High", "Medium", "Low", "Lower", "Lowest"]
+  parameter.parameter6enumMap = ["Highest": 1, "Higher": 2, "High": 3, "Medium": 4, "Low": 5, "Lower": 6, "Lowest": 7]
+  parameter.parameter6default = "Medium"
+  parameter.parameter6size = 1
+  
+  parameter.parameter7name = "LedDispMode"
+  parameter.parameter7title = "LED display mode"
+  parameter.parameter7description = ""
+  parameter.parameter7type = "enum"
+  parameter.parameter7options = ["LED off when Temperature report/Motion active",
+                       "LED blink when Temperature report/Motion active",
+                       "LED blink when Motion active/LED off when Temperature report"]
+  parameter.parameter7enumMap = ["LED off when Temperature report/Motion active": 1,
+                       "LED blink when Temperature report/Motion active": 2,
+                       "LED blink when Motion active/LED off when Temperature report": 3]
+  parameter.parameter7default = "LED off when Temperature report/Motion active"
+  parameter.parameter7size = 1
+  
+  parameter.parameter8name = "RetryTimes"
+  parameter.parameter8title = "Motion notification retry times"
+  parameter.parameter8description = ""
+  parameter.parameter8type = "number"
+  parameter.parameter8options = "0..10"
+  parameter.parameter8enumMap = []
+  parameter.parameter8default = 3
+  parameter.parameter8size = 1
+  
+  return parameter."parameter${num}${text}"
+}
+
+def wakeUpIntervalUpdate(interval) {
+  def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+  
+  if (value != null) {
+    if (state."${getWakeUpInfo("name")}" != value) {
+      state."${getWakeUpInfo("name")}" = value
+      state."${getWakeUpInfo("name")}Refresh" = true
+    }
+  }
+}
+
+def wakeUpIntervalCheck(interval) {
+  def value = interval?.find { it.key == "${getWakeUpInfo("name")}" }?.value
+  
+  if (value != null) {
+    if (state."${getWakeUpInfo("name")}" == value) {
+      if (state."${getWakeUpInfo("name")}Refresh" == true) {    
+        state."${getWakeUpInfo("name")}Refresh" = false
+        logInfo "wake up interval *** check passed"
+      }
+    }
+  }
+}
+
+def getWakeUpInfo(text) {
+  def wakeUp = [:]
+  
+  wakeUp.name = "wakeUpInterval"
+  wakeUp.title = "Wake up interval [Hours]"
+  wakeUp.description = ""
+  wakeUp.type = "number"
+  wakeUp.range = "1..4660"
+  wakeUp.default = 24
+  
+  return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
```

Comment:
CommandClassVersions are not obligatory. You may use zwave.parse(description) instead.

Reply:
I do not modify it, since I would like to specify the command class version for used.